File: C#/hellodeath!.cs
using System;

namespace CursedHelloWorld
{
    class Program
    {
        static void Main(string[] args)
        {
            Action<object> w = Console.WriteLine;
            Func<string, string> s = x => x;
            w(s("\x48\x65\x6c\x6c\x6f\x2c\x20\x57\x6f\x72\x6c\x64\x21"));
        }
    }
}

File: GO/hellodeath!.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	helloWorld := strings.Map(func(r rune) rune {
		return r
	}, "\x48\x65\x6C\x6C\x6F\x20\x57\x6F\x72\x6C\x64\x21")

	fmt.Println(helloWorld)
}